core/templates/dev/head/pages/exploration_editor/RouterServices.js

```diff
@@ -44,14 +44,20 @@ oppia.factory('routerService', [
       return $location.path();
     }, function(newPath, oldPath) {
       if (newPath === '') {
-        $location.path(oldPath);
+        if (oldPath === '/') {
+          $window.history.go(-1);
+        } else {
+          $location.path(oldPath);
+        }
         return;
       }
-
       if (!oldPath) {
         // This can happen when clicking on links whose href is "#".
         return;
       }
+      if (newPath === '/' && oldPath.includes('/gui/')) {
```

Comment:
I thought that pressing the "create" button on the library page takes you to the dashboard page (i.e. the logged-in homepage), which then redirects you to the exploration editor (although it doesn't do that automatically if you have access to the collection editor too, since then you need to choose which option to pick). So if you do that, then if you click "back", I _think_ you should end up on the dashboard, not the library.

However, you are right about the /create/exploration_id --> /create/exploration_id#/ --> /create/exploration_id#/gui/(statename) being an issue with the editor. Can we use replaceState for that one? So the overall "back button" history for a user in this scenario would be something like:

    - /library
    - /dashboard
    - /create/exploration_id#/gui/(statename)